### ISSUE TITLE
pr and wthpr don't corelate

### ISSUE THREAD

**Issue by pkp24** [alphaparrot/ExoPlaSim#39]
I'm running a simulation of an eyeball planet with TR85 and am getting super high wthpr rainfall, yet pr rainfall is much much much less. Is there something wrong with the output of one of them or is it really pouring down 60 feet per day?

![pr_in_MOST_SNAP 00018](https://github.com/user-attachments/assets/3cb11eea-17f6-4510-b5b8-33946e00f8d9)
![wthpr_in_MOST_SNAP 00018](https://github.com/user-attachments/assets/f18b0fe3-8d25-4eee-8141-902eb0967812)

**Comment by alphaparrot** [comment]
It definitely seems like something is wrong here. Have you done any transformations to the output of either prior to plotting, like an attempted unit conversion? `wthpr` is simply the average sum of convective and large-scale precipitation (not snow) at timesteps and locations where there is both land and surface temperature above 273.15K, multiplied by 8.64*10^7 to convert from m/s to mm/day. The pr output variable is simply the average sum of convective and large-scale precipitation (excluding snow) at each timestep and location. If I multiply your max pr by 8.64*10^7 I get about 160 mm/day as the highest possible value we should be seeing in wthpr (incidentally the max wthpr value appears to be roughly 160 times that value--maybe something is being squared?). I don't see anything in the code that would account for a discrepancy here.

**Comment by pkp24** [comment]
I haven't, this is just me taking the output and putting it straight into panapoly. Currently the wthpr is now double what it was, this is still within the first 40 years so I'm assuming the snow on the land is contributing heavily to the rainfall. But I feel like that shouldn't make a difference for the two parameters to be so far from each other.

**Comment by alphaparrot** [comment]
Oh I see the problem. This is from a snapshot file. Looks like I have a minor bug in the weathpr output for snapshot files--it's saving the time-averaged value, only it's not dividing by the number of timesteps since the last time-averaged output (though it does get reset at each time-averaged output write). That's both incorrect and annoying. You can salvage existing output (sort of) by figuring out how many timesteps have passed since the last time-averaged output write (MOST_OUT instead of MOST_SNAP) before the snapshot in question and dividing by that, and I'm guessing that'll get you to the right ballpark, though it'll still be time-averaged, and not snapshot data.

For future outputs, try ExoPlaSim 3.4.2--I think that should work; I've got a test run going right now to verify. Although snapshot output for weatherable precipitation may not be that interesting _anyway_; for most applications involving weathering calculations it's probably better to use time-averaged output anyway (it'll give you a fuller picture of what's happening over time). 

I'll mark this closed once I confirm that the new output is working as expected.

**Comment by alphaparrot** [comment]
Yep, looks like that fixed it, judging by output from the first year of a quick T21 Earth run:
![weathpr_fix](https://github.com/user-attachments/assets/b0711efb-9932-4ec8-b7f1-56d94930623a)
